src/cmd/flux-jobspec.py

```diff
@@ -176,6 +181,180 @@ def slurm_jobspec(args):
     )
 
 
+def positive_nonzero_int(string):
+    value = int(string)
+    if value <= 0:
+        msg = "{} is not positive"
+        raise argparse.ArgumentTypeError(msg)
+    return value
+
+
+def parse_fsd(s):
+    m = re.match(r".*([smhd])$", s)
+    n = float(s[:-1] if m else s)
+    unit = m.group(1) if m else "s"
+
+    if unit == "m":
+        seconds = timedelta(minutes=n).total_seconds()
+    elif unit == "h":
+        seconds = timedelta(hours=n).total_seconds()
+    elif unit == "d":
+        seconds = timedelta(days=n).total_seconds()
+    else:
+        seconds = n
+    return seconds
+
+
+def resource_walker(res):
+    while res is not None and len(res) > 0:
+        if type(res) is dict:
+            yield res
+            res = res.get("with", None)
+        elif type(res) is list:
+            res = res[0]
```

Comment:
Whilst working on the Jobspec class refactor, I noticed that this function only works on jobspec with lists of len == 1.  If any of the resources have multiple children, it won't traverse the extra children.

FWIW, the function I ended up with is:
```python
        def walk_helper(res_list):                                                                                                                                                                                                                                                                    
            for resource in res_list:                                                                                                                                                                                                                                                                 
                yield resource                                                                                                                                                                                                                                                                        
                children = resource.get("with", [])                                                                                                                                                                                                                                                   
                # TODO: convert to `yield from` after dropping 2.7                                                                                                                                                                                                                                    
                for x in walk_helper(children):                                                                                                                                                                                                                                                       
                    yield x
```